src/java.base/share/classes/sun/security/rsa/RSAPrivateCrtKeyImpl.java

```diff
@@ -276,16 +271,32 @@ protected long getNativePtr() {
         byte[] dQ_2c   = dQ.toByteArray();
         byte[] qInv_2c = qInv.toByteArray();
 
-        nativeRSAKey = nativeCrypto.createRSAPrivateCrtKey(n_2c,n_2c.length, d_2c, d_2c.length, e_2c, e_2c.length,
+        newNativePtr = nativeCrypto.createRSAPrivateCrtKey(n_2c,n_2c.length, d_2c, d_2c.length, e_2c, e_2c.length,
                 p_2c, p_2c.length, q_2c, q_2c.length,
                 dP_2c, dP_2c.length, dQ_2c, dQ_2c.length, qInv_2c, qInv_2c.length);
-        return nativeRSAKey;
+        
+        return newNativePtr;
+    }
+
+    protected long getNativePtr() {
+        Long ptr = keyQ.poll();
+        if (ptr == null) {
+            return RSAPrivateKey_generate();
+        }
+        return ptr;
+    }
+
+    protected void returnNativePtr(Long ptr) {
+        if (ptr != null) {
+            keyQ.add(ptr);
+        } 
     }
 
     @Override
     public void finalize() {
-        if (nativeRSAKey != 0x0 && nativeRSAKey != -1) {
-            nativeCrypto.destroyRSAKey(nativeRSAKey);
+        Long itr = null;
```

Comment:
Please remove redundant initialization (javac will ensure it's assigned a value before use).